### PR TITLE
Remove KeyInfo and use standard library to parse hex strings

### PR DESCRIPTION
**Pull request by jmwilson**
Addresses #2. Key parsing can be made shorter using `std::from_chars`, which requires C++17. Since this is a small utility without dependencies, I moved the standard to C++17, which makes a lot of features available.

As a functionality change, key checksums are no longer computed or printed, and I feel like that is a security improvement. Logging anything that allows key material to be inferred is bad.

### CODE REVIEW

**Comment by jonasstein** [comment]
Is this solution better than https://stackoverflow.com/a/69327772/1749675 ?

Personally I like the idea of C++ 17 or even 20 as minimum requirement. 
But we should decide so first in (https://github.com/scsitape/stenc/issues/49)

**Comment by jmwilson** [comment]
Well, that SO question is for going the other direction (binary -> text representation).

This type of task is easily susceptible to https://en.wikipedia.org/wiki/Code_golf, but I think this is a pretty good solution using modern C++.

File: src/main.cpp [hunk 1]
```diff
@@ -262,11 +292,9 @@ int main(int argc, char **argv) {
     exit(EXIT_SUCCESS);
   }
 
-  Keyinfo ki{};
   if (drvOptions.cryptMode != CRYPTMODE_OFF) {
     if (keyFile == "") {
-      std::string p1 = "01";
-      std::string p2 = "02";
+      std::string p1, p2;
```
Comment (jonasstein) [inline src/main.cpp:297]: somewhere I read one should

- initialize
- write one variable per line, not more

```
std::string p1 {};
std::string p2 {};
```

Comment (jmwilson) [reply src/main.cpp:297]: One declaration per line is good, this is part of C++ core guidelines https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rl-dcl

The default constructor of string creates an empty string, so `string s {};` is needlessly verbose.

**Pull request merged**